test/unitTests.js

```diff
@@ -65,31 +85,49 @@ describe('Correct email and correct password log in test', function() {
        assert.notEqual(response, null);
         })
+    });
+
+
+// Unit Tests for Home Page
+const assert = require('assert');
+const city = "Toronto"
+const groupSize = 20
+const startDate = new Date("2020-11-05")
+const endDate = new Date("2021-01-05")
 
+describe('Recommend Attractions Test', function() {
+ it('Recommend Attractions should return 4', async function() {
+       const response = await RecommendController.recommend(city,groupSize,startDate,endDate)
+       assert.equal(response.length,4);
+        })
     });
 
-// This unit test is for the sign up page and authentication controller
-// It makes sure that the users are placed in the database
-describe('Sign up test', function(){
-    it('Signup should place users in database and return the same length as user', async function(){
-        // this will pass the information collected in signup to the authentication controller 
-        const response = await AuthenticationController.signup(email,name,password)
-        // a user is found given the constants and parameters for users
-        const user = await Users.findOne({
-            where:{
-                Email:email,
-                Name: name,
-                Password: password,
-                Age: null,
-                Gender: null,
-                Instagram_Username: null,
-                Preferred_Activity_Type: null
-            }
+
+
+
+// Unit Tests for Onboarding Page
+const correctOnboardingEmail = "rniazi@uwaterloo.ca"
+const incorrectOnboardingEmail = "rushan@incorrect.ca"
+const acceptableAge = 20
+const unacceptableAge = 11
+const acceptableGender = "Male"
+const acceptableInstagramUsername = "rushan_niazi"
+const acceptableActivity = "Family"
+
+
+describe('Onboarding Test for Correct email, age, gender, instagram username, and preferred activity type', function() {
+ it('Onboarding test should return true with all attribute data being correct', async function() {
+       const response = await AuthenticationController.onboardingTest(correctOnboardingEmail,acceptableAge,acceptableGender,acceptableInstagramUsername,acceptableActivity)
+       assert.equal(response, null);
         })
-        // the length of the user returned in authentication controller should be the same as the length of the user returned from the users file
-        // if the length of these are the same, the signup function is working properly
-        assert.equal(response.length,user.length);
-    })
-});
     });
 
+describe('Onboarding Test for Incorrect email and age, however correct gender, instagram username, and preferred activity type', function() {
+ it('Onboarding test should return false since data for email and age are incorrect', async function() {
+       const response = await AuthenticationController.onboardingTest(incorrectOnboardingEmail,unacceptableAge,acceptableGender,acceptableInstagramUsername,acceptableActivity)
+       assert.equal(response, null);
+        })
+    });
```

Comment:
Should we test for other unacceptable values as well? Why are we only testing for unacceptable email and age?
Also, could we test with just unacceptable age and a correct email and one with acceptable age but unacceptable email?

Reply:
Great idea, I am hopefully planning on extending these unit tests to consider all edge cases such are you mentioned